ricks/serializers.py

```diff
@@ -1,11 +1,20 @@
-from rest_framework.serializers import ModelSerializer
+from rest_framework import serializers
 from .models import Rick
+from morties.models import Morty
 
 
-class RickSerializer(ModelSerializer):
+class RickSerializer(serializers.ModelSerializer):
     """
         Serializer for Ricks
     """
+    paired_morty_universe = serializers.SerializerMethodField()
+
     class Meta:
         model = Rick
         fields = "__all__"
+
+    def get_paired_morty_universe(self, rick_instance):
+        try:
+            return rick_instance.paired_morty.universe
+        except Morty.DoesNotExist:
```

Comment:
Volta pra ver a segunda resposta daquele link the eu te passei. Veja como eles tem uma parte em que eles consegue verificar algo do tipo `if model.related_model is None`. 

Pra fazer isso, eles fizeram uma coisa difirente da sua solução. É isso que você tem que mudar, pra não ter que ficar usando try..except nem `hasattr`.

Reply:
Cara, confesso que não entendi muito bem. Você se refere à essa parte aqui?

class Place(models.Model):
    name = models.CharField(max_length=50)
    address = models.CharField(max_length=80)

class Restaurant(models.Model):  # The class where the one-to-one originates
    place = models.OneToOneField(Place, blank=True, null=True)
    serves_hot_dogs = models.BooleanField()
    serves_pizza = models.BooleanField()
    
… to see if a Restaurant has a Place, we can use the following code:

r = Restaurant(serves_hot_dogs=True, serves_pizza=False)
r.save()
if r.place is None:
    print "Restaurant has no place!"
    
Restaurant has no place!
    
Mas em seguida ele fala:

To see if a Place has a Restaurant, it's important to understand that referencing the restaurant property on an instance of Place raises a Restaurant.DoesNotExist exception if there is no corresponding restaurant. This happens because Django performs a lookup internally using QuerySet.get(). For example:

p2 = Place(name='Ace Hardware', address='1013 N. Ashland')
p2.save()
p2.restaurant
Traceback (most recent call last):
    ...
DoesNotExist: Restaurant matching query does not exist.

Daí fiquei sem entender como fazer o check sem vir uma exception (talvez seja só limitação minha... 😅)

Reply:
Exato, ele tem 2 models que estão relacionados, `Place` e `Restaurant`.

Do jeito como foi feito, ele consegue fazer a comparação `if restaurant.place is None`, mas não consegue fazer `if place.restaurant is None`, por que retorna um erro.

No nosso código, só queremos ver se `rick.morty is None`, mas não o contrário. Ou seja, seria legal se no nosso código a coisa funcionasse como no primeiro caso explicado lá.

Você consegue descobrir por que um funciona e outro não? E daí descobrindo isso, consegue fazer funcionar igual no nosso caso?